src/routes.js

```diff
@@ -44,4 +45,25 @@ router.post('/coffeegrams', async ctx => {
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
+router.post('/users/:username', async ctx => {
+  var user = await User.findOne({ username: ctx.params.username });
+  if (!user) {
+    throw new NotFound();
+  }
+  if (user.id !== ctx.state.currentUser.id) {
+    ctx.request.flash('error', "You\'re not allowed to edit someone else\'s profile");
+    return ctx.redirect('back');
+  }
+
+  var { bio } = ctx.request.body;
+  clean(bio);
```

Comment:
This is still a no-op. The `clean()` function returns a **new** string, it doesn't change the existing string "in-place". Strings, Booleans and Integers are called "primitives" or "immutable", which means they cannot be changed in-place. In contrast, Objects (like Objects, Arrays, Functions, etc) are "mutable" types, and can be modified in place.